[2025-03-22T11:09:26.459+0800] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-03-22T11:09:26.464+0800] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: transport_dag.transform_bus_data_task manual__2025-03-22T03:03:49.790223+00:00 [queued]>
[2025-03-22T11:09:26.467+0800] {taskinstance.py:2614} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: transport_dag.transform_bus_data_task manual__2025-03-22T03:03:49.790223+00:00 [queued]>
[2025-03-22T11:09:26.467+0800] {taskinstance.py:2867} INFO - Starting attempt 2 of 2
[2025-03-22T11:09:26.472+0800] {taskinstance.py:2890} INFO - Executing <Task(PythonOperator): transform_bus_data_task> on 2025-03-22 03:03:49.790223+00:00
[2025-03-22T11:09:26.475+0800] {standard_task_runner.py:72} INFO - Started process 93805 to run task
[2025-03-22T11:09:26.478+0800] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'transport_dag', 'transform_bus_data_task', 'manual__2025-03-22T03:03:49.790223+00:00', '--job-id', '156', '--raw', '--subdir', 'DAGS_FOLDER/main.py', '--cfg-path', '/var/folders/08/pmnk96px7x94djx_0kg9f2zh0000gn/T/tmpv9prddc9']
[2025-03-22T11:09:26.480+0800] {standard_task_runner.py:105} INFO - Job 156: Subtask transform_bus_data_task
[2025-03-22T11:09:26.495+0800] {task_command.py:467} INFO - Running <TaskInstance: transport_dag.transform_bus_data_task manual__2025-03-22T03:03:49.790223+00:00 [running]> on host melvins-macbook-pro-2.local
[2025-03-22T11:09:26.516+0800] {taskinstance.py:3134} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='transport_dag' AIRFLOW_CTX_TASK_ID='transform_bus_data_task' AIRFLOW_CTX_EXECUTION_DATE='2025-03-22T03:03:49.790223+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-03-22T03:03:49.790223+00:00'
[2025-03-22T11:09:26.517+0800] {taskinstance.py:732} INFO - ::endgroup::
[2025-03-22T11:09:26.546+0800] {taskinstance.py:3313} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/airflow/dags/main.py", line 105, in transform_bus_data
    bus_routes = pd.DataFrame(ti.xcom_pull(task_ids='extract_bus_routes_task', key='bus_routes'))
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/pandas/core/frame.py", line 886, in __init__
    raise ValueError("DataFrame constructor not properly called!")
ValueError: DataFrame constructor not properly called!
[2025-03-22T11:09:26.551+0800] {taskinstance.py:1226} INFO - Marking task as FAILED. dag_id=transport_dag, task_id=transform_bus_data_task, run_id=manual__2025-03-22T03:03:49.790223+00:00, execution_date=20250322T030349, start_date=20250322T030926, end_date=20250322T030926
[2025-03-22T11:09:26.558+0800] {taskinstance.py:341} INFO - ::group::Post task execution logs
[2025-03-22T11:09:26.559+0800] {standard_task_runner.py:124} ERROR - Failed to execute job 156 for task transform_bus_data_task (DataFrame constructor not properly called!; 93805)
Traceback (most recent call last):
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3006, in _run_raw_task
    return _run_raw_task(
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 274, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3161, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 3185, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 768, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/taskinstance.py", line 734, in _execute_callable
    return ExecutionCallableRunner(
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/models/baseoperator.py", line 424, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/airflow/dags/main.py", line 105, in transform_bus_data
    bus_routes = pd.DataFrame(ti.xcom_pull(task_ids='extract_bus_routes_task', key='bus_routes'))
  File "/Users/melvinang/Documents/NUS/Y3S2/IS3107/transport_project/IS3107-Group-8/local_airflow/.airflow/lib/python3.9/site-packages/pandas/core/frame.py", line 886, in __init__
    raise ValueError("DataFrame constructor not properly called!")
ValueError: DataFrame constructor not properly called!
[2025-03-22T11:09:26.584+0800] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-03-22T11:09:26.591+0800] {taskinstance.py:3901} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-03-22T11:09:26.592+0800] {local_task_job_runner.py:245} INFO - ::endgroup::
